crates/sui-benchmark/src/workloads/adversarial.rs

```diff
@@ -214,3 +234,53 @@ impl Workload<dyn Payload> for AdversarialWorkload {
             .collect()
     }
 }
+
+struct AdversarialPayloadArgs {
+    pub fn_name: String,
+    pub args: Vec<CallArg>,
+}
+
+fn get_payload_args(
+    payload_type: AdversarialPayloadType,
+    protocol_config: &ProtocolConfig,
+) -> AdversarialPayloadArgs {
+    match payload_type {
+        AdversarialPayloadType::ObjectsSize => AdversarialPayloadArgs {
+            fn_name: "create_shared_objects".to_owned(),
+            args: [
+                // TODO: Raise this. Using a smaller value here as full value locks up local machine
+                (NUM_OBJECTS / 10).into(),
+                // TODO: Raise this. Using a smaller value here as full value locks up local machine
+                (protocol_config.max_move_object_size() / 10).into(),
+            ]
+            .to_vec(),
+        },
+        AdversarialPayloadType::EventSize => AdversarialPayloadArgs {
+            fn_name: "emit_events".to_owned(),
+            args: [
+                // TODO: Raise this. Using a smaller value here as full value locks up local machine
+                (protocol_config.max_num_event_emit() / 10).into(),
+                // TODO: Raise this. Using a smaller value here as full value errs
+                (protocol_config.max_event_emit_size() / 3).into(),
+            ]
+            .to_vec(),
+        },
+        AdversarialPayloadType::DynamicFieldsCount => AdversarialPayloadArgs {
+            fn_name: "add_dynamic_fields".to_owned(),
+            args: [protocol_config
+                .object_runtime_max_num_cached_objects()
+                .into()]
+            .to_vec(),
+        },
+        AdversarialPayloadType::EventCount => AdversarialPayloadArgs {
+            fn_name: "emit_events".to_owned(),
+            args: [
+                // TODO: Raise this. Using a smaller value here as full value locks up local machine
+                protocol_config.max_num_event_emit().into(),
+                // Intentionally emitting small objects as we're only checking the count
```

Comment:
I would say: let's emit events that are as big as possible (do we have an event size limit?). More generally, whenever we have a choice In these adversarial payloads, I would suggest going with the choice that is hardest for the system to handle (bigger objects, more objects, bigger events, ...) without bumping into a limit somewhere.

Reply:
We quickly run out of gas when emitting events that are both large and plentiful, making us unable to hit either limit hence why I separated it into two tests: one that emits max number of small events, and one that emits a low number of max-sized events.

Reply:
Condensed both now